src/main/java/org/jabref/gui/actions/CleanupAction.java

```diff
@@ -39,6 +43,26 @@ public CleanupAction(BasePanel panel, JabRefPreferences preferences) {
     }
 
     @Override
+    public void action() {
+        init();
+        Platform.runLater(() -> {
```

Comment:
Please use the `runInJavaFXThread` method of `DefaultTaskExectuor` (we want to separate calls that are needed because we still have interaction with legacy Swing and "real" async operations in JavaFX)

On a second thought, I'm actually wondering if this call to `runLater` is even necessary since the action should be invoked through the menu which is already on the JavaFX thread.